docs/testplan-style.md

```diff
@@ -0,0 +1,243 @@
+*   `feature/interface/singleton/ate_tests/singleton_test/singleton_test.go`
+    implements the issue.
+*   `feature/interface/aggregate/ate_tests/aggregate_test/README.md` documents
+    the test plan for the issue
+    [RT-5.2 Aggregate Interface](https://github.com/openconfig/featureprofiles/issues/112).
+*   `feature/interface/aggregate/ate_tests/aggregate_test/aggregate_test.go`
+    implements the issue.
+
+## Test Plan Template
+
+The test plan in `README.md` is generally structured like this:
+
+```
+# RT-5.1: Singleton Interface
+
+## Summary
+
+[Insert a 1-3 sentence description of the test.]
+
+## Procedure
+
+[Write an ordered list of steps.  Each step may be given a
+descriptive name.]
+
+1. Step 1
+2. Step 2
+3. ...
+
+## Config Parameter Coverage
+
+[Write a list of OpenConfig paths which are expected to be included.]
+
+*   /interfaces/interface/config/name
+*   /interfaces/interface/config/description
+*   ...
+
+## Telemetry Parameter Coverage
+
+[Write a list of OpenConfig paths which are expected to be included.]
+
+*   /interfaces/interface/state/oper-status
+*   /interfaces/interface/state/admin-status
+*   ...
+```
+
+## Development Approach for Tests
+
+### Vendor Neutrality
+
+Test plans should use a vendor-neutral mechanism to configure the DUT, e.g. in
+the OpenConfig public model or other OpenConfig protocols such as gNMI, gRIBI,
+and gNOI. It should depend only on the behavior of the open specification.
+
+### Device/Topology Neutrality
+
+Tests are to be built against a minimal topology that consists of a single
+device (referred to as DUT) against a reference implementation (referred to as
+an ATE) of routing protocols and data plane functionality.
+
+Each test plan must choose from one of the following testbeds and adhere to it
+for the entirety of the test plan:
+
+*   atedut\_2: two interconnected ports between ATE and DUT.
+*   atedut\_4: four interconnected ports between ATE and DUT.
+*   atedut\_12: 4 singleton ports and 8 breakout ports interconnected between
+    ATE and DUT. Ondatra testbed models the logical ports, but the test is
+    responsible for activating the breakout.
+*   dutdut: four interconnected ports between two DUTs.
+
+Test plans must not hardcode assumptions about the presence of specific hardware
+components (e.g. switch chip, controller card, line card). If needed, it may
+accept a "device profile" which parameterizes the expectations about the device.
+If the test plan is about a feature that does not make sense for a given device
+profile, the test should be skipped. For example, a test about controller card
+switchover should be skipped if the device profile indicates that the device is
+expected to have only one controller card, rather than skipping based on the
+number of controller cards detected from the device.
+
+### Functional Testing vs. Fuzzing
+
+Initially, the test plan will consist of manually identified functional features
```

Comment:
Does this need to be the case? Is there any reason that one cannot define a test plan that is fuzzing? Is this solely because of where the current tests that are defined are coming from rather than what could be contributed by anyone to this repository?

Reply:
To be more concrete, this feels like a priority call that could be made by each author of a new test. For some folks, fuzzing might be the most important thing :-)